spec/services/preserved_object_handler_confirm_version_spec.rb

```diff
@@ -0,0 +1,335 @@
+          end
+        end
+      end
+
+      context 'db update error' do
+        context 'ActiveRecordError' do
+          let(:result_code) { PreservedObjectHandlerResults::DB_UPDATE_FAILED }
+          let(:incoming_version) { 2 }
+          let(:results) do
+            allow(Rails.logger).to receive(:log)
+            # FIXME: couldn't figure out how to put next line into its own test
+            expect(Rails.logger).to receive(:log).with(Logger::ERROR, /#{db_update_failed_prefix_regex_escaped}/)
+
+            po = instance_double("PreservedObject")
+            allow(PreservedObject).to receive(:find_by).with(druid: druid).and_return(po)
+            pc = instance_double("PreservedCopy")
+            allow(PreservedCopy).to receive(:find_by).and_return(pc)
+            allow(pc).to receive(:version).and_return(2)
+            allow(pc).to receive(:status)
+            allow(pc).to receive(:last_audited=)
+            allow(pc).to receive(:last_checked_on_storage=)
+            allow(pc).to receive(:changed?).and_return(true)
+            allow(pc).to receive(:save!).and_raise(ActiveRecord::ActiveRecordError, 'foo')
+            allow(po).to receive(:current_version).and_return(2)
+            po_handler.confirm_version
+          end
+
+          context 'DB_UPDATE_FAILED error' do
+            it 'prefix' do
+              expect(results).to include(a_hash_including(result_code => a_string_matching(db_update_failed_prefix_regex_escaped)))
+            end
+            it 'specific exception raised' do
+              expect(results).to include(a_hash_including(result_code => a_string_matching('ActiveRecord::ActiveRecordError')))
+            end
+            it "exception's message" do
+              expect(results).to include(a_hash_including(result_code => a_string_matching('foo')))
+            end
+          end
+        end
+      end
+
+      it 'calls PreservedCopy.save! if the existing record is altered' do
+        po = instance_double(PreservedObject)
+        pc = instance_double(PreservedCopy)
+        status = PreservedCopy::EXPECTED_VERS_NOT_FOUND_ON_STORAGE_STATUS
+        allow(PreservedObject).to receive(:find_by).with(druid: druid).and_return(po)
+        allow(po).to receive(:current_version).and_return(1)
+        allow(po).to receive(:save!)
+        allow(PreservedCopy).to receive(:find_by).with(preserved_object: po, endpoint: ep).and_return(pc)
+        allow(pc).to receive(:version).and_return(1)
+        allow(pc).to receive(:status).and_return(status)
+        allow(pc).to receive(:last_audited=)
+        allow(pc).to receive(:last_checked_on_storage=)
+        allow(pc).to receive(:changed?).and_return(true)
+        allow(pc).to receive(:save!)
+        po_handler.confirm_version
+        expect(po).not_to have_received(:save!)
+        expect(pc).to have_received(:save!)
+      end
+      it 'calls PreservedObject.touch and PreservedCopy.touch if the existing record is NOT altered' do
```

Comment:
looks like title should actually be `'calls PreservedCopy.touch (but not PreservedObject.touch) if the existing record is NOT altered'` (or similar)

Reply:
good catch - meant to change that test description.